### PR TITLE
root: 6.26.08 -> 6.26.10; add xrootd support and workaround

### PR DESCRIPTION
**Pull request by ShamrockLee**
###### Description of changes

*   Update to 6.26.10 and go back to `fetchurl`.
*   Enable xrootd support and apply the `[DY]LD_LIBRARY_PATH` workaround before we have #200830 gets ready.
    That is also how the upstream-provided `./setxrd.sh` works. Considering the importance of this feature, it would be worthwhile to apply the workaround first.
*   Use `stdenv.hostPlatform.is*` instead of `stdenv.is*`. This brings us closer toward cross-compilation.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/physics/hepmc3/default.nix [hunk 1]
```diff
@@ -16,9 +24,15 @@ stdenv.mkDerivation rec {
...
 
-  nativeBuildInputs = [ cmake ];
-  buildInputs = [ root_py ]
-    ++ lib.optional withPython python;
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  buildInputs = [
+    nlohmann_json # Needed since root 6.26.10
```
Comment (veprbl) [inline pkgs/development/libraries/physics/hepmc3/default.nix:32]: Move to `root.propagatedBuildInputs`?

File: pkgs/applications/science/misc/root/default.nix [hunk 2]
```diff
@@ -114,8 +114,8 @@ stdenv.mkDerivation rec {
     python.pkgs.numpy
     tbb
   ]
-  ++ lib.optionals (!stdenv.isDarwin) [ libX11 libXpm libXft libXext libGLU libGL ]
-  ++ lib.optionals (stdenv.isDarwin) [ Cocoa CoreSymbolication OpenGL ]
+  ++ lib.optionals (!stdenv.hostPlatform.isDarwin) [ libX11 libXpm libXft libXext libGLU libGL ]
+  ++ lib.optionals (stdenv.hostPlatform.isDarwin) [ Cocoa CoreSymbolication OpenGL ]
```
Comment (veprbl) [inline pkgs/applications/science/misc/root/default.nix:118]: What is this for? hostPlatform is the default lookup, and it's not like we are going to implement cross compilation.

Comment (ShamrockLee) [reply pkgs/applications/science/misc/root/default.nix:118]: > hostPlatform is the default lookup

Glad to know that. I'll drop that commit.

File: pkgs/development/libraries/physics/hepmc3/default.nix (hunk 1 above)
Comment (ShamrockLee) [reply pkgs/development/libraries/physics/hepmc3/default.nix:32]: Specifying it in `root.propagatedBuildInputs` alone doesn't eliminate the CMake error when building hepmc3.

Should we inherit the `propagatedBuildInputs` from root in hepmc/default.nix?

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 205394` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>xfitter</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>python310Packages.hepmc3</li>
    <li>python39Packages.hepmc3</li>
    <li>root</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by veprbl** [comment]
@GrahamcOfBorg build xfitter

**Comment by veprbl** [comment]
This one is odd:
```output
> /nix/store/cwivsnh1p591x7yydynrz0xc1zcq54p3-binutils-2.39/bin/ld: /nix/store/alp33wf3d30qpnw98r92xw8qwv7jj6lb-root-6.26.10/lib/libTree.so: undefined reference to `std::__istream_extract(std::istream&, char*, long)@GLIBCXX_3.4.29'
> /nix/store/cwivsnh1p591x7yydynrz0xc1zcq54p3-binutils-2.39/bin/ld: /nix/store/alp33wf3d30qpnw98r92xw8qwv7jj6lb-root-6.26.10/lib/libCore.so: undefined reference to `std::__throw_bad_array_new_length()@GLIBCXX_3.4.29'
```

**Pull request merged**